--- gexiv2/meson.build.orig	2019-05-03 17:01:42.000000000 -0700
+++ gexiv2/meson.build	2019-05-03 17:01:45.000000000 -0700
@@ -53,7 +53,7 @@
                  [version_header] +
                  enum_sources,
                  include_directories : include_directories('..'),
-                 version: '2.0.0',
+                 version: '3.0.0',
                  dependencies : [gobject, exiv2, gio],
                  install : true)
 
